### DISCUSSION TITLE
Hello!

### DISCUSSION THREAD

**Discussion by OOHehir** [LIFsCode/ELOC-3.0#20] [General]
Hello all, 

I'm new to this project, I'm looking forward to helping out & hope I can be of some use!

This is my first time using PlatformIO (I'm using it via Visual Studio Code  on a Linux machine). I think I have a handle on it but any pointers are appreciated. 

I've just created a PR which includes:
- A suggested change for a path separator, it was breaking the build on my machine, not sure if the case on others
- Initial commit of TensorFlow libraries & test models. It was a bit of a struggle bringing them in, lots of inter-dependencies, I eventually ended up stripping them out of another library, details outlined in lib/tflite-micro/README.md

As detailed in the PR the issue with this commit is memory, or lack of it.

My C++ skills are only moderate so if I'm doing/ creating anything that could be better please let me know!

Thanks,

Owen

**Comment by EDsteve** [comment]
Hi Owen,

sorry that i haven't introduced you yet. But you have already done that nicely with a PR. Didn't know that you are already work on our firmware implementation. Great :) So welcome to the ELOC project @OOHehir:) 

**Quick introduction** so everybody knows what is going on and who is who:
Fabi (@LIFsCode) is a friend of mine from Germany, helping us writing our firmware in his free time. He works with firmware and electronics professionally and has improved our firmware significantly. As we know; Free time is never enough. Specially in summer time :) Therefore we hire freelancers for additional jobs and that's how i found Owen :)

Owen (Expert in IoT from Ireland) is working together with Shyam right now (a prof of the Curtin University Australia), who is providing us with a Tensor flow model. Owen will convert it to a TFlite model and implement it into our firmware. He has a lot of experience with IoT in general (hard and software) and is hopefully with us along the way with LoRa or other implementations in the future.

Can't say much about the PR changes. But i can give a comment to the **memory issue:**
Fabi has sent me [details](https://github.com/LIFsCode/ELOC-3.0/issues/16) in how to properly make the external PSRAM work, which should open up more space in the IRAM. Unfortunately the Freelancer i hired for it does not reply any more so i will have to find a new one. So that issue should be dealt with sooner or later :)

Fabi, how do you like to deal with Owens PRs? Into master or AI branch?

Cheeers
ED

**Reply by OOHehir** [reply]
@EDsteve many thanks. As mentioned in a direct email I should be able to look at the PSRAM issue, might take a bit of time to get up to speed with. If any work on this can be briefly outlined it might be useful.

Regards,

Owen

**Comment by LIFsCode** [comment]
Hi Owen, great to have you on the project.

Currently I'm not at home so I have limited access to the code sorry. I will be back on 31 july and process your PR then. I would first keep all AI & PSRAM related stuff on a separate AI branch an merge it to master once everything is functional.

Concerning the memory issue: 
Main reason for the limited memory (IRAM) is the fixed partitioning which is done by the platformIO/espressif framework which uses parts of the internal RAM as instruction RAM for code which is performance critical (I.e. Bluetooth stack & PSRAM access driver code)
Since we are pushing it to the limits of memory we would need more control over the partitioning and placent of code so we can either

- increase the partition for IRAM
- move code which is for our application not performance critical to other segments

I tried to outline a bit in the other issue ED mentioned: https://github.com/LIFsCode/ELOC-3.0/issues/16

By the way I just landed in your country Owen ;-)  I'm heading for a boat trip along the Shannon river the next week.

Greetings
Fabian

**Comment by OOHehir** [comment]
Fabian,

Welcome to Ireland! You certainly won't have any issues with extreme
temperatures, just extreme rainfall! The weather has been very unsettled
recently & looks to continue into next week unfortunately.

No problem about keeping things separate, perhaps I'll close the current PR
& recreate when it's in a working state.

I'll take a look at the link & try to implement a change that'll get the
code into a working state. Ill add to the issue in github as I go & if
you've any suggestions/ requirements regarding code placement (e.g.
critical/ performance stuff) please let me know.

Regards,

Owen

**Reply by LIFsCode** [reply]
@OOHehir : BTW a bit off topic :) I'm in Dublin today and tomorrow...do you maybe have any good suggestions for a good place to get a beer or good food?

**Comment by OOHehir** [comment]
@LIFsCode 

If I was you I'd avoid Temple Bar, a bit of a tourist trap!

### Food:
Trocadero Restaurant Dublin (More traditional fare)
No. 4, St Andrew's St, Dublin 2, D02 PD30

Red Torch Ginger Dublin 2
14/15 St Andrew's St, Dublin, D02 HP58

Saba Restaurant
7 William St S, Dublin 2, D02 CV65

Xian Street Food Dublin
28 Anne St S, Dublin, D02 DX39

### Pubs:
Kehoes Pub
9 Anne St S, Dublin, D02 NY88

Mulligan's
8 Poolbeg St, Dublin 2, DO2TK71

The Palace Bar
21 Fleet St, Temple Bar, Dublin 2, D02 H950

### After drinks kebab:
Zaytoon
15 Parliament St, Temple Bar, Dublin, D02 FW60

Hope you enjoy your visit!

**Reply by LIFsCode** [reply]
Cool thanks a lot
